src/common/CustomIconButton/CustomIconButton.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types';
+import { EuiButton } from '@elastic/eui';
+
+const CustomIconButton = ({
+  iconType, href, target, ...rest
+}) => {
+  if (process.env.NODE_ENV === 'test') {
+    return (
+      <EuiButton
+        href={href}
+        target={target}
+      >
+        {rest.children}
```

Comment:
Define the `children` prop instead of getting it from `rest`. Doing `...rest` it's a bad practice but it's acceptable in this case to avoid defining all the supported props in `EuiButton`, however, if you need a prop that's coming as part of `rest` in this component, then you have to define it explicitly otherwise it will be hidden the component contract.

Reply:
Besides, `...rest` should be provided to `EuiButton` as `<EuiButton  href={href}  target={target} {...rest} />

Reply:
ummm I've noticed something more and probably need verbal explanation :P Look at `href` and `target`, are you using them inside this component as part of a condition or logic? For me the answer is no. Those props are passed to `EuiButton` without touching them so why would you need to define them as props?  If you pass `rest` as I describe in the previous comment you're all good. Here you should only define `iconType` and `children` because you are explicitly using them in the component.

Reply:
Fixed